start.go

```diff
@@ -1,139 +1,32 @@
-is a directory with a specification file named "` + specConfig + `" and a root
-filesystem.
-
-The specification file includes an args parameter. The args parameter is used
-to specify command(s) that get run when the container is started. To change the
-command(s) that get executed on start, edit the args parameter of the spec. See
-"runc spec --help" for more explanation.`,
-	Flags: []cli.Flag{
-		cli.StringFlag{
-			Name:  "bundle, b",
-			Value: "",
-			Usage: `path to the root of the bundle directory, defaults to the current directory`,
-		},
-		cli.StringFlag{
-			Name:  "console",
-			Value: "",
-			Usage: "specify the pty slave path for use with the container",
-		},
-		cli.BoolFlag{
-			Name:  "detach,d",
-			Usage: "detach from the container's process",
-		},
-		cli.StringFlag{
-			Name:  "pid-file",
-			Value: "",
-			Usage: "specify the file to write the process id to",
-		},
-		cli.BoolFlag{
-			Name:  "no-subreaper",
-			Usage: "disable the use of the subreaper used to reap reparented processes",
-		},
-		cli.BoolFlag{
-			Name:  "no-pivot",
-			Usage: "do not use pivot root to jail process inside rootfs.  This should be used whenever the rootfs is on top of a ramdisk",
-		},
-	},
-	Action: func(context *cli.Context) error {
-		bundle := context.String("bundle")
-		if bundle != "" {
-			if err := os.Chdir(bundle); err != nil {
-				return err
-			}
-		}
-		spec, err := loadSpec(specConfig)
+	Description: `The start command signals the container to start the user's defined process.`,
+	Action: func(context *cli.Context) {
+		container, err := getContainer(context)
 		if err != nil {
-			return err
-		}
-
-		notifySocket := os.Getenv("NOTIFY_SOCKET")
-		if notifySocket != "" {
-			setupSdNotify(spec, notifySocket)
-		}
-
-		if os.Geteuid() != 0 {
-			return fmt.Errorf("runc should be run as root")
+			fatal(err)
 		}
-
-		status, err := startContainer(context, spec)
-		if err == nil {
-			// exit with the container's exit status so any external supervisor is
-			// notified of the exit with the correct exit status.
-			os.Exit(status)
+		status, err := container.Status()
+		if err != nil {
+			fatal(err)
 		}
-		return err
-	},
-}
-
-func init() {
-	if len(os.Args) > 1 && os.Args[1] == "init" {
-		runtime.GOMAXPROCS(1)
-		runtime.LockOSThread()
-	}
-}
-
-var initCommand = cli.Command{
-	Name:  "init",
-	Usage: `initialize the namespaces and launch the process (do not call it outside of runc)`,
-	Action: func(context *cli.Context) error {
-		factory, _ := libcontainer.New("")
-		if err := factory.StartInitialization(); err != nil {
-			// as the error is sent back to the parent there is no need to log
-			// or write it to stderr because the parent process will handle this
-			os.Exit(1)
+		if status == libcontainer.Created {
+			if err := container.Signal(libcontainer.InitContinueSignal); err != nil {
+				fatal(err)
+			}
 		}
-		panic("libcontainer: container init failed to exec")
```

Comment:
`panic` is probably too strong, but I expect we want an error here to cover the "that container is in {state}, but you can only start 'Created' containers” case (where opencontainers/runtime-spec#384 is currently [calling for an error][1]).

[1]: https://github.com/opencontainers/runtime-spec/pull/384/files#diff-b84a8d65d8ed53f4794cd2db7e8ea731R86